concordium-node/src/bin/cli.rs

```diff
@@ -152,9 +153,33 @@ async fn main() -> anyhow::Result<()> {
         None
     };
 
-    if let Some(ref import_path) = conf.cli.baker.import_path {
+    // Out-of-band catch-up
+    if let Some(ref import_blocks_from) = conf.cli.baker.import_blocks_from {
+        let import_path =
+            data_dir_path.to_path_buf().join(concordium_node::configuration::CATCHUP_FILE);
+        let from = match Url::parse(import_blocks_from) {
+            Ok(import_url) => {
+                info!("Downloading catch-up files");
+                let res: Result<(), Box<dyn std::error::Error + Send + Sync>> = async {
+                    let mut file = std::fs::File::create(import_path.clone())?;
+                    let response = reqwest::get(import_url).await?;
+                    let mut content = Cursor::new(response.bytes().await?);
+                    std::io::copy(&mut content, &mut file)?;
```

Comment:
Instead of loading the entire file into memory you should stream it into a file. Something like

```rust
response.bytes_stream().try_for_each(|bytes| async { file.write_all(&bytes) }).await?
```

Reply:
https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_for_each